### PR TITLE
Use event.series.count for pod events, default to 1 if event count is not set

### PR DESCRIPTION
**Pull request by AObuchow**
### What does this PR do?
Adds a wrapper function to get the workspace deployment pod event count, instead of just using `event.count`.

When checking workspace deployment pod events, ` event.series.count` is checked first (as `event.count` is [deprecated](https://kubernetes.io/docs/reference/using-api/deprecation-guide/#event-v125)). If `event.series.count` isn't set, the fallback is to check `event.count` if it is set. Otherwise, the event count is assumed to be 1.

Draft PR for now as I'm still waiting for my cluster to start up to test this out. 

### What issues does this PR fix or reference?
#987

### Is it tested? How?
1. Deploy DWO to an OpenShift 4.11 or 4.12 cluster i.e `make docker install`
2. Apply the following devworkspace, which requests a large amount of memory:
```YAML
kind: DevWorkspace
apiVersion: workspace.devfile.io/v1alpha2
metadata:
  name: theia-next-high-mem
spec:
  started: true
  template:
    projects:
      - name: web-nodejs-sample
        git:
          remotes:
            origin: "https://github.com/che-samples/web-nodejs-sample.git"
    components:
      - name: theia
        plugin:
          uri: https://che-plugin-registry-main.surge.sh/v3/plugins/eclipse/che-theia/latest/devfile.yaml
          components:
            - name: theia-ide
              container:
                env:
                  - name: THEIA_HOST
                    value: 0.0.0.0
                memoryRequest: 32Gi
                memoryLimit: 64Gi
                cpuRequest: 500m
                cpuLimit: 2000m
    commands:
      - id: say-hello
        exec:
          component: theia-ide
          commandLine: echo "Hello from $(pwd)"
          workingDir: ${PROJECTS_ROOT}/project/app

```
3. Check DWO's logs for an error message similar to the following: 
```JSON
{
   "level":"info",
   "ts":1669920891.8467853,
   "logger":"controllers.DevWorkspace",
   "msg":"DevWorkspace failed to start: Detected unrecoverable event FailedScheduling: 0/6 nodes are available: 3 node(s) had untolerated taint {node-role.kubernetes.io/master: }, 6 Insufficient memory. preemption: 0/6 nodes are available: 3 No preemption victims found for incoming pod, 3 Preemption is not helpful for scheduling..",
   "Request.Namespace":"devworkspace-controller",
   "Request.Name":"theia-next-high-mem",
   "devworkspace_id":"workspacebf274e95a04e43d5"
}
```
4. Do a `kubectl get dw -n $NAMESPACE` and check that the workspace has failed and the INFO points to the `FailedScheduling` event.

```
NAME                  DEVWORKSPACE ID             PHASE      INFO
theia-next-high-mem   workspacebf274e95a04e43d5   Failing    Detected unrecoverable event FailedScheduling: 0/6 nodes are available: 3 node(s) had untolerated taint {node-role.kubernetes.io/master: }, 6 Insufficient memory. preemption: 0/6 nodes are available: 3 No preemption victims found for incoming pod, 3 Preemption is not helpful for scheduling..

```

Note that this error message differs from the expected error message for the same workspace in https://github.com/devfile/devworkspace-operator/pull/978. This is because pod events are checked before pod status conditions.

### PR Checklist

- [ ] E2E tests pass (when PR is ready, comment `/test v8-devworkspace-operator-e2e, v8-che-happy-path` to trigger)
    - [ ] `v8-devworkspace-operator-e2e`: DevWorkspace e2e test
    - [ ] `v8-che-happy-path`: Happy path for verification integration with Che

### CODE REVIEW

File: pkg/library/status/check.go [hunk 1]
```diff
@@ -181,3 +181,12 @@ func checkPodConditions(pod *corev1.Pod) (msg string) {
 	}
 	return ""
 }
+
+func getEventCount(event corev1.Event) int32 {
```
Comment (amisevsk) [inline pkg/library/status/check.go:185]: Could you include a quick doc explaining the reasoning for this function (i.e. how we came to the end of this confusing road on event counts :P )?

**Review by amisevsk — APPROVED** [review]
LGTM

File: pkg/library/status/check.go (hunk 1 above)
Comment (ibuziuk) [reply pkg/library/status/check.go:185]: would be also nice to add a link to the OpenShift issue

Comment (AObuchow) [reply pkg/library/status/check.go:185]: +1 will do :)

**Review by ibuziuk — APPROVED** [review]

**Pull request merged**